src/practices/Go/GoGitignoreCorrectlySetPractice.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { GoGitignoreCorrectlySetPractice } from './GoGitignoreCorrectlySetPractice';
+import { gitignoreContent } from '../../detectors/__MOCKS__/Go/gitignoreContent.mock';
+import { PracticeEvaluationResult } from '../../model';
+import {
+  TestContainerContext,
+  createTestContainer,
+} from '../../inversify.config';
+
+const basicGitignore = `
+*.exe
+*.exe~
+*.dll
+*.so
+*.dylib
+*.test
+*.out
+vendor/
+`;
+
+describe('GoGitignoreCorrectlySetPractice', () => {
+  let practice: GoGitignoreCorrectlySetPractice;
+  let containerCtx: TestContainerContext;
+
+  beforeAll(() => {
+    containerCtx = createTestContainer();
+    containerCtx.container
+      .bind('GoGitignoreCorrectlySetPractice')
+      .to(GoGitignoreCorrectlySetPractice);
+    practice = containerCtx.container.get('GoGitignoreCorrectlySetPractice');
+  });
+
+  afterEach(async () => {
+    containerCtx.virtualFileSystemService.clearFileSystem();
+    containerCtx.practiceContext.fileInspector!.purgeCache();
+  });
+
+  it('Returns practicing if the .gitignore is set correctly', async () => {
+    containerCtx.virtualFileSystemService.setFileSystem({
+      '.gitignore': gitignoreContent,
+    });
+
+    const evaluated = await practice.evaluate(containerCtx.practiceContext);
+    expect(evaluated).toEqual(PracticeEvaluationResult.practicing);
+  });
+
+  it('Returns notPracticing if there the .gitignore is NOT set correctly', async () => {
+    containerCtx.virtualFileSystemService.setFileSystem({
+      '.gitignore': '...',
+    });
+
+    const evaluated = await practice.evaluate(containerCtx.practiceContext);
+    expect(evaluated).toEqual(PracticeEvaluationResult.notPracticing);
+    expect(practice.data.details).not.toBeUndefined();
+  });
+
+  it('Returns unknown if there is no fileInspector', async () => {
+    const evaluated = await practice.evaluate({
+      ...containerCtx.practiceContext,
+      ...{ root: { fileInspector: undefined } },
+    });
+    expect(evaluated).toEqual(PracticeEvaluationResult.unknown);
+  });
+
+  describe('Fixer', () => {
+    afterEach(async () => {
+      containerCtx.virtualFileSystemService.clearFileSystem();
+    });
+
+    it('Does not change correct .gitignore', async () => {
+      const gitignore = `${basicGitignore}\npackage-lock.json\n`;
```

Comment:
There is no `package-lock.json` in Go lang. Please, change it.

Reply:
we need to simply remove it or add any other file likt that of `package-lock.json`.

Reply:
You can just remove the `package-lock.json`